### PR TITLE
De-duplicate on press page

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

De-duplication of stories on containers is moved to facia-press when building 'lite' and 'full' fronts. This enables lite versions of fronts to be used, where it was impossible to derive deduplicated containers without a surplus of container stories.

## What is the value of this and can you measure success?

Faster Facia front responses.

## Test in CODE?

Yes

### CODE REVIEW

**Comment by QuarpT** [comment]
I've removed the de-duplication admin tools. They aren't feasible since the de-duplicated content can't be derived from a pressed front anymore. 

Doesn't appear these get used looking at request logs. Nice to remove complexity from the codebase.

File: facia-press/app/frontpress/PressedCollectionVisibility.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  lazy val affectsDuplicates: Boolean = Container.affectsDuplicates(pressedCollection.collectionType, pressedCollection.curatedPlusBackfillDeduplicated)
+  lazy val affectedByDuplicates: Boolean = Container.affectedByDuplicates(pressedCollection.collectionType, pressedCollection.curatedPlusBackfillDeduplicated)
+
+  lazy val deduplicateAgainst: Seq[String] = {
+    if (affectsDuplicates)
+      pressedCollection
+        .curatedPlusBackfillDeduplicated
+        .take(visible)
+        .filter(_.participatesInDeduplication)
+        .map(_.header.url)
```
Comment (TBonnin) [inline facia-press/app/frontpress/PressedCollectionVisibility.scala]: best to use `_.properties.webUrl` I think

File: common/app/layout/slices/Container.scala [hunk 2]
```diff
@@ -39,6 +39,21 @@ object Container extends Logging {
     }
   }
 
+  def affectsDuplicates(collectionType: String, items: Seq[PressedContent]): Boolean = {
```
Comment (TBonnin) [inline common/app/layout/slices/Container.scala]: items doesn't seem to be used

File: common/app/layout/slices/Container.scala [hunk 3]
```diff
@@ -39,6 +39,21 @@ object Container extends Logging {
...
+    resolve(collectionType) match {
+      case Fixed(fixedContainer) if !fixedContainer.isSingleton => true
+      case Dynamic(_) => true
+      case _ => false
+    }
+  }
+
+  def affectedByDuplicates(collectionType: String, items: Seq[PressedContent]): Boolean = {
```
Comment (TBonnin) [inline common/app/layout/slices/Container.scala]: same here

**Review by TBonnin — CHANGES_REQUESTED** [review]

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 4]
```diff
@@ -303,7 +300,11 @@ trait FapiFrontPress extends Logging {
       pressedCollections <- Response.traverse(collectionIds.map(generateCollectionJsonFromFapiClient))
       seoWithProperties <- Response.Async.Right(getFrontSeoAndProperties(path))
     } yield seoWithProperties match {
-      case (seoData, frontProperties) => PressedPageVersions.fromPressedCollections(path, seoData, frontProperties, pressedCollections)
+      case (seoData, frontProperties) =>
+        val dedupliatedCollections = PressedCollectionVisibility.deduplication(pressedCollections)
+          .map(_.pressedCollectionVersions)
+          .toList
```
[the remark this answers is not in the document]
Comment (QuarpT) [reply facia-press/app/frontpress/FapiFrontPress.scala]: This is facia press

[the remark this answers is not in the document]
Comment (TBonnin) [reply facia-press/app/frontpress/FapiFrontPress.scala]: Oups I misread then 😊

File: common/app/layout/Front.scala [hunk 5]
```diff
@@ -375,13 +290,12 @@ object Front extends implicits.Collections {
...
     def faciaContainers(allConfigs: Seq[((ContainerDisplayConfig, CollectionEssentials), Container)],
                         context: ContainerLayoutContext,
-                        seenTrails: Set[TrailUrl] = Set.empty,
                         index: Int = 0,
                         accumulation: Vector[FaciaContainer] = Vector.empty): Seq[FaciaContainer] = {
       allConfigs.toList match {
         case Nil => accumulation
         case ((config, collection), container) :: remainingConfigs =>
-          val (newSeen, newItems, _) = deduplicate(seenTrails, container, collection.items)
+          val newItems = collection.items.distinctBy(_.header.url)
```
[the remark this answers is not in the document]
Comment (QuarpT) [reply common/app/layout/Front.scala]: There won't be unless there are duplicates in a single collection

File: common/app/layout/slices/Container.scala (hunk 2 above)
Comment (QuarpT) [reply common/app/layout/slices/Container.scala]: You're right!

File: common/app/layout/slices/Container.scala (hunk 3 above)
Comment (QuarpT) [reply common/app/layout/slices/Container.scala]: 👍

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 6]
```diff
@@ -178,14 +178,11 @@ trait FapiFrontPress extends Logging {
       val storyCountVisible = Container.storiesCount(collection.collectionConfig.collectionType, curated ++ backfill).getOrElse(storyCountMax)
       val hasMore = storyCountVisible < storyCountMax
 
-      PressedCollectionVersions(
-        pressCollection(collection, curated, backfill, treats, storyCountVisible, hasMore),
-        pressCollection(collection, curated, backfill, treats, storyCountMax, hasMore)
-      )
+      val pressedCollection = pressCollection(collection, curated, backfill, treats, storyCountMax, hasMore)
```
[the remark this answers is not in the document]
Comment (QuarpT) [reply facia-press/app/frontpress/FapiFrontPress.scala]: No, it gets set later on. I will default it to false

File: common/app/layout/Front.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (TBonnin) [reply common/app/layout/Front.scala]: ok

File: facia-press/app/frontpress/PressedCollectionVisibility.scala (hunk 1 above)
Comment (QuarpT) [reply facia-press/app/frontpress/PressedCollectionVisibility.scala]: It used to be de-duplicated against `header.url`

File: facia-press/app/frontpress/PressedCollectionVisibility.scala [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import model.PressedCollectionVersions
+import model.facia.PressedCollection
+import model.pressed.PressedContent
+
+
+case class PressedCollectionVisibility(pressedCollection: PressedCollection, visible: Int) extends implicits.Collections {
+  lazy val affectsDuplicates: Boolean = Container.affectsDuplicates(pressedCollection.collectionType, pressedCollection.curatedPlusBackfillDeduplicated)
+  lazy val affectedByDuplicates: Boolean = Container.affectedByDuplicates(pressedCollection.collectionType, pressedCollection.curatedPlusBackfillDeduplicated)
+
+  lazy val deduplicateAgainst: Seq[String] = {
```
[the remark this answers is not in the document]
Comment (QuarpT) [reply facia-press/app/frontpress/PressedCollectionVisibility.scala]: Can do

File: common/app/layout/Front.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (QuarpT) [reply common/app/layout/Front.scala]: Also, do we still want de-duplication for index pages? They don't use pressed fronts

File: facia-press/app/frontpress/PressedCollectionVisibility.scala (hunk 1 above)
Comment (TBonnin) [reply facia-press/app/frontpress/PressedCollectionVisibility.scala]: nevermind then.

**Review by TBonnin — APPROVED** [review]

**Pull request merged**